addrmgr/addrmanager_test.go

```diff
@@ -108,80 +108,66 @@ func lookupFunc(host string) ([]net.IP, error) {
 	return nil, errors.New("not implemented")
 }
 
-func TestStartStop(t *testing.T) {
-	n := New("teststartstop", lookupFunc)
-	n.Start()
-	if err := n.Stop(); err != nil {
-		t.Fatalf("Address Manager failed to stop: %v", err)
-	}
+// addAddressByIP is a convenience function that adds an address from an ip
+// address string and port, rather than a wire.NetAddress.
+func (a *AddrManager) addAddressByIP(addr string, port uint16) {
+	ip := net.ParseIP(addr)
+	na := wire.NewNetAddressIPPort(ip, port, 0)
+	a.AddAddress(na, na)
 }
 
-func TestAddAddressByIP(t *testing.T) {
-	fmtErr := fmt.Errorf("")
-	addrErr := &net.AddrError{}
-	var tests = []struct {
-		addrIP string
-		err    error
-	}{
-		{
-			someIP + ":8333",
-			nil,
-		},
-		{
-			someIP,
-			addrErr,
-		},
-		{
-			someIP[:12] + ":8333",
-			fmtErr,
-		},
-		{
-			someIP + ":abcd",
-			fmtErr,
-		},
-	}
-
-	dir, err := ioutil.TempDir("", "testaddressbyip")
+// TestStartStop tests the behavior of the address manager when it is started
+// and stopped.
+func TestStartStop(t *testing.T) {
```

Comment:
This is probably not the PR to this in but since you are working on the address manager you should consider converting it to leverage contexts and ditch start/stop process.